notes

```diff
@@ -117,16 +118,47 @@ generate_name() {
 }
 
 new_note() {
-    local note_name="$*"
-    if [[ $note_name == "" ]]; then
-         note_name="$(generate_name)"
+    local template_used=false
+    local note_name=""
+
+    if [[ "$1" == "-t" ]]; then
+        template_used=true
+        shift
+    fi
+
+    if [[ "$#" -eq 0 ]]; then
+        note_name="$(generate_name).$NOTES_EXT"
+    elif [[ "$#" -eq 1 ]]; then
+        note_name="$1"
+    else
+        note_name="$2.$NOTES_EXT"
     fi
 
     if echo "$note_name" | grep "/$" &> /dev/null; then
         note_name="${note_name}/$(generate_name)"
     fi
 
-    mkdir -p "$(dirname "$notes_dir/$note_name")"
+    local note_dir="$(dirname "$notes_dir/$note_name")"
+    mkdir -p "$note_dir"
+
+    if "$template_used"; then
+        local template_folder="$notes_dir/$TEMPLATES_DIR"
+
+        if [[ ! -d "$template_folder" ]]; then
+            printf "Template folder not found: $template_folder\n"
+            exit 1
+        fi
+
+        local template_name="$1"
+
+        if [[ ! "$template_name" == *.$NOTES_EXT ]]; then
+            template_name="$template_name.$NOTES_EXT"
+        fi
+
+        local template_path="$template_folder/$template_name"
+        local note_path="$notes_dir/$note_name"
+        cp "$template_path" "$note_path"
```

Comment:
We should check if the template actually exists, and fail with a nice message if it doesn't